src/assets/src/validation.ts

```diff
@@ -76,3 +77,59 @@ export function validateString (value: string, schema: StringSchema, showRemaini
     }
     return {'transformedValue': transformedValue, 'isInvalid': isInvalid, 'messages': messages};
 }
+
+
+export interface MeetingTypesValidationResult {
+    isInvalid: boolean;
+    messages: ReadonlyArray<string>;
+}
+
+export function validateMeetingTypes (value: Set<string>, queue?: QueueHost): MeetingTypesValidationResult {
+    let messages = Array();
+
+    const noTypesSelected = value.size === 0;
+    if (noTypesSelected) messages.push('You must select at least one allowed meeting type.');
+
+    let existingMeetingConflict = false;
+    if (queue) {
+        const uniqueMeetingTypes = new Set(queue!.meeting_set.map(m => m.backend_type));
+        let conflictingTypes = Array();
+        uniqueMeetingTypes.forEach(uniqueMeetingType => {
+            if (!value.has(uniqueMeetingType)) conflictingTypes.push(uniqueMeetingType);
+        });
+        if (conflictingTypes.length > 0) {
+            existingMeetingConflict = true;
+            messages.push(
+                'You cannot disallow the following meeting types until the meetings ' +
+                'using them have been removed from the queue: ' +
+                conflictingTypes.join(', ')
+            );
+        }
+    }
+    return { isInvalid: (noTypesSelected || existingMeetingConflict), messages: messages };
+}
+
+
+// Utility wrappers for both validating and updating React state using hook setter functions
+
+type StringValidationResultSetter = React.Dispatch<React.SetStateAction<undefined | ValidationResult>>;
+type MeetingTypesValidationResultSetter = React.Dispatch<React.SetStateAction<undefined | MeetingTypesValidationResult>>
+
+export function validateAndSetStringResult (
+    someValue: string,
+    schema: StringSchema,
+    resultSetter: StringValidationResultSetter,
+    showRemaining?: boolean
+): ValidationResult {
+    const validationResult = validateString(someValue, schema, !!showRemaining);
+    resultSetter(validationResult);
+    return validationResult;
+};
```

Comment:
This is good, but you could also make this a hook, e.g. `useValidatedStringState`, that returns `[string, setAndValidateString, clear]` where `setAndValidateString` is something like this and `clear` is `setString(undefined)` (because I see you invoked that above).

Reply:
I'd like to think about this a little more. It's an interesting idea, but I'm not sure what more we'd be getting with the hook version...